posthog/models/kafka_engine_dlq/sql.py

```diff
@@ -9,23 +11,31 @@
 # to make sure we're not silently dropping events and so we
 # can have full oversight over the end-to-end ingestion pipeline.
 
-KAFKA_ENGINE_DLQ_BASE_SQL = """
+TTL_POLICY = ttl_period("timestamp", 8)  # 8 weeks
+
+KAFKA_ENGINE_DLQ_BASE_SQL = (
+    """
 CREATE TABLE IF NOT EXISTS {table} ON CLUSTER '{cluster}'
 (
+    timestamp DateTime,
     topic VARCHAR,
     partition Int64,
     offset Int64,
     raw VARCHAR,
     error VARCHAR
 )
 ENGINE = {engine}
-ORDER BY (topic, partition, offset)
+PARTITION BY toYYYYMM(timestamp)
+ORDER BY (topic, partition, offset, toStartOfHour(timestamp))
```

Comment:
Note: `toStartOfHour` will not help the ORDER BY since offset always increments. It's also not relevant for the engine.

```suggestion
ORDER BY (topic, toStartOfDay(timestamp), partition, offset)
```